src/duty/state/state.service.ts

```diff
@@ -91,4 +102,33 @@ export class StateService {
       },
     });
   }
+
+  //https://github.com/ChainSafe/lodestar/blob/stable/packages/beacon-node/src/api/impl/beacon/state/utils.ts
+  public getValidatorStatus(validator: any, currentEpoch: Epoch): ValStatus {
+    // pending
+    if (validator.activationEpoch > currentEpoch) {
+      if (validator.activationEligibilityEpoch === FAR_FUTURE_EPOCH) {
+        return ValStatus.PendingInitialized;
+      } else if (validator.activationEligibilityEpoch < FAR_FUTURE_EPOCH) {
+        return ValStatus.PendingQueued;
+      }
+    }
+    // active
+    if (validator.activationEpoch <= currentEpoch && currentEpoch < validator.exitEpoch) {
+      if (validator.exitEpoch === FAR_FUTURE_EPOCH) {
+        return ValStatus.ActiveOngoing;
+      } else if (validator.exitEpoch < FAR_FUTURE_EPOCH) {
+        return validator.slashed ? ValStatus.ActiveSlashed : ValStatus.ActiveExiting;
+      }
+    }
+    // exited
+    if (validator.exitEpoch <= currentEpoch && currentEpoch < validator.withdrawableEpoch) {
+      return validator.slashed ? ValStatus.ExitedSlashed : ValStatus.ExitedUnslashed;
+    }
+    // withdrawal
+    if (validator.withdrawableEpoch <= currentEpoch) {
+      return validator.effectiveBalance !== 0 ? ValStatus.WithdrawalPossible : ValStatus.WithdrawalDone;
+    }
+    throw new Error('ValidatorStatus unknown');
+  }
 }
```

Comment:
I suggest removing extra checkings and rephrasing code of the function this way:
```
public getValidatorStatus(validator: any, currentEpoch: Epoch): ValStatus {
    // pending
    if (validator.activationEpoch > currentEpoch) {
      if (validator.activationEligibilityEpoch === Infinity) {
        return ValStatus.PendingInitialized;
      }

      return ValStatus.PendingQueued;
    }

    // active
    if (validator.exitEpoch > currentEpoch) {
      if (validator.exitEpoch === Infinity) {
        return ValStatus.ActiveOngoing;
      }

      return validator.slashed ? ValStatus.ActiveSlashed : ValStatus.ActiveExiting;
    }

    // exited
    if (validator.withdrawableEpoch > currentEpoch) {
      return validator.slashed ? ValStatus.ExitedSlashed : ValStatus.ExitedUnslashed;
    }

    // withdrawal
    return validator.effectiveBalance !== 0 ? ValStatus.WithdrawalPossible : ValStatus.WithdrawalDone;
  }
```
Seems more readable to me.

And also remove the `FAR_FUTURE_EPOCH` constant. It looks redundant for this code.

Reply:
the function is just copy of https://github.com/ChainSafe/lodestar/blob/778a171ce026c29cfb22cb7e4634f0632c35f0e4/packages/beacon-node/src/api/impl/beacon/state/utils.ts#L67
I would like not to change something here just to be sure that it works as expected. 
`FAR_FUTURE_EPOCH ` is the term from consensus spec, so I'd like to keep it like the const just to be aligned with the spec